module4/lessons/grouping_aggregating.md

```diff
@@ -0,0 +1,657 @@
+---
+layout: page
+title: Grouping and Aggregating
+module: 4
+---
+
+## Learning Goals
+
+- Understand what an aggregate function does
+- Understand what the group SQL statement does
+- Understand how the group statement relates to aggregate functions
+- Write aggregate/group queries in SQL
+- Write aggregate/group queries in ActiveRecord
+- Write aggregate/group/joins queries in ActiveRecord
+
+## Vocabulary
+
+- SQL Group
+- SQL Aggregate Function
+- ActiveRecord Calculation
+
+## Recording
+
+[This recording](https://youtu.be/shUhwkiT2Jg) will cover the conceptual parts of the lesson, plus some bloopers at the beginning 😁
+
+## Aggregate Functions
+
+PostgresSQL comes with some handy built-in aggregate functions. From the [Postgresql Docs](https://www.postgresql.org/docs/14/functions-aggregate.html): `Aggregate functions compute a single result from a set of input values`. Basically, aggregate functions will do some math for us.
+
+Take a minute to look through the [docs](https://www.postgresql.org/docs/14/functions-aggregate.html) and see what the DB can do for you.
+
+While we are talking about functions built in to Postgresql, it is important to note that nearly all relational database systems include some form of aggregating, for example [MySQL](https://dev.mysql.com/doc/refman/8.0/en/group-by-modifiers.html).
+
+### SQL Aggregates
+
+We'll be working with Set List on the `many-to-many-complete` branch.  Let's run `rails dbconsole` in SetList to open up a connection to our Postgres Database. Let's find the average length of all songs:
```

Comment:
Perhaps here would be a good spot to link the repo if we're open to that idea?

Reply:
Link has been added!